---
title: "Assignments May"
author: "Mio Shibata"
date: "2018/6/13"
output: html_document
---

##1st section
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
```

## 
We are going to read in this data, then test your knowledge of they key dplyr functions select and filter. We are also going to review two different classes: data frames and vectors.
1. Read in the msleep_ggplot2.csv file with the function read.csv and use the function class to determine what type of object is returned.
```{r}
url <- "https://raw.githubusercontent.com/genomicsclass/dagdata/master/inst/extdata/msleep_ggplot2.csv"
msleep <- read.csv(url)
class(msleep)
```

2. Now use the filter function to select only the primates.
How many animals in the table are primates? 
First, I used the filter() to select only the primates and put the data.frame into msleep2.
The colnames() shows the name of columns and head() can show from 1st to 10th row.
The first column is "name", and the result of hed() implies the first column has the animal
the function of nrow() can show the number of rows ,then the answer is shown below.
```{r}
msleep2<-filter(msleep, order %in% c( "Primates"))

colnames(msleep2)
head(msleep2)
nrow(msleep2)
```

3. What is the class of the object you obtain after subsetting the table to only include primates?
```{r}
class(msleep2)
```

4. Now use the select function to extract the sleep(total) for the primates. 
What classis thisobject? Hint: use %>% to pipe the results of the filter function to select.

```{r}
msleep3<-msleep2 %>% 
         select(name, sleep_total) %>% 
         filter  
msleep3
    
```

5. Now we want to calculate the average amount of sleep for primates (the average of the numbers computed above). One challenge is that the mean function requires a vector so, if we simply apply it to the output above, we get an error. Look at the help file for unlist and
use it to compute the desired average.
```{r}
mean(msleep3$sleep_total)
```

6. For the last exercise, we could also use the dplyr summarize function. We have not introduced this function, but you can read the help file and repeat exercise 5, this time using just filter and summarize to get the answer.

```{r}
msleep %>% 
    summarise(avg_sleep = mean(sleep_total), 
              min_sleep = min(sleep_total),
              max_sleep = max(sleep_total),
              total = n())
```

##2nd section
We showed how to calculate a Chi-square test from a table. Here we will show how to generate the table from data which is in the form of a dataframe, so that you can then perform an association test to see if two columns have an enrichment (or depletion) of shared occurrences.

Download the https://studio.edx.org/c4x/HarvardX/PH525.1x/asset/assoctest.csv file into your R working directory, and then read it into R: d = read.csv("assoctest.csv")
```{r}
library(downloader) 
url <- "https://studio.edx.org/c4x/HarvardX/PH525.1x/asset/assoctest.csv"
filename <- "assoctest.csv"
download(url, destfile=filename)

```

1. This dataframe reflects the allele status (either AA/Aa or aa) and the case/control status for 72 individuals. Compute the Chi-square test for the association of genotype with case/control status (using the table function and the chisq.test function). Examine the table to see if there appears to be an association.

```{r}
dat <- read.csv(filename)
head(dat)
```
Assume that 0 stands for (AA/Aa) or 1 does(aa) , and 0 does (control) or 1 does (cases),

```{r}
allele<-factor(dat$allele,levels=c("0","1"))
disease<-factor(dat$case,levels=c("0","1"))
table(allele)
tab <- table(allele,disease)
tab
p=mean(disease=="1")#1 means "cases"
p
chisq.test(tab)$p.value

```
 What is the X-squared statistic?
This statistic is used when we want to test the null hypothesis that there is relation between two variants. The test above is that one variant is the allele status and the other is case/control status. 


2. Compute Fisher’s exact test fisher.test for the sametable. 
```{r}
fisher.test(tab)
```
What is the p-value?
The p-value is the criteria to determine the 
If the result of p values is less than 0.05, the null hypothesis which claims that two variants are dependent is very rare and it should be rejected.
But the result is not less that 0.05 ,then the null hypothesis can't be rejected and the null hypothesis is still remained.
